src/db/models/Integrations.js

```diff
@@ -1,23 +1,315 @@
+   * @param {string} doc.formData.formEmail - TODO: need more elaborate documentation
+   * @param {string} doc.formData.userEmailTitle - TODO: need more elaborate documentation
+   * @param {string} doc.formData.userEmailContent - TODO: need more elaborate documentation
+   * @param {Email[]} doc.formData.adminEmails - TODO: need more elaborate documentation
+   * @param {string} doc.formData.adminEmailTitle - TODO: need more elaborate documentation
+   * @param {string} doc.formData.adminEmailContent - TODO: need more elaborate documentation
+   * @param {string} doc.formData.thankContent - TODO: need more elaborate documentation
+   * @param {string} doc.formData.redirectUrl - Form redirectUrl on submit
+   * @param {string} args.mainDoc - Integration main document object
+   * @param {string} args.mainDoc.name - Integration name
+   * @param {string} args.mainDoc.brandId - Integration brand id
+   * @param {string} args.mainDoc.formId - Form id related to this integration
+   * @return {Promise} returns form integration document promise
+   * @throws {Exception} throws Exception if formData is notSupplied
+   */
+  static createFormIntegration({ formData, ...mainDoc }) {
+    const doc = this.generateFormDoc(mainDoc, formData);
+
+    if (Object.keys(formData || {}).length === 0) {
+      throw new Error('formData must be supplied');
+    }
+
+    return this.create(doc);
+  }
+
+  /**
+   * Update form integration
+   * @param {string} _id integration id
+   * @param {Object} args.formData - FormData object
+   * @param {string} doc.formData.loadType - Load types for the embedded form
+   * @param {string} doc.formData.successAction - TODO: need more elaborate documentation
+   * @param {string} doc.formData.formEmail - TODO: need more elaborate documentation
+   * @param {string} doc.formData.userEmailTitle - TODO: need more elaborate documentation
+   * @param {string} doc.formData.userEmailContent - TODO: need more elaborate documentation
+   * @param {Email[]} doc.formData.adminEmails - TODO: need more elaborate documentation
+   * @param {string} doc.formData.adminEmailTitle - TODO: need more elaborate documentation
+   * @param {string} doc.formData.adminEmailContent - TODO: need more elaborate documentation
+   * @param {string} doc.formData.thankContent - TODO: need more elaborate documentation
+   * @param {string} doc.formData.redirectUrl - Form redirectUrl on submit
+   * @param {string} args.mainDoc - Integration main document object
+   * @param {string} args.mainDoc.name - Integration name
+   * @param {string} args.mainDoc.brandId - Integration brand id
+   * @param {string} args.mainDoc.formId - Form id related to this integration
+   * @return {Promise} returns null promise
+   */
+  static updateFormIntegration(_id, { formData, ...mainDoc }) {
+    const doc = this.generateFormDoc(mainDoc, formData);
+
+    return this.update({ _id }, { $set: doc }, { runValidators: true });
```

Comment:
return updated integration
await this.update ....
return this.findOne({ _id })